inductiva/resources/machines_base.py

```diff
@@ -65,7 +65,7 @@ def __init__(
         provider = machine_types.ProviderType(provider)
         self.provider = provider.value
 
-        if data_disk_gb <= 0:
+        if data_disk_gb <= 0 and provider != machine_types.ProviderType.LOCAL:
```

Comment:
So, if our machine is LOCAL I can send data_disk_gb=-1 ?

Reply:
The server has no knowledge of the disk size of a local machine, that variable has no meaning if the machine is local.

Reply:
So, if we are changing the behavior of a class based on the type, shouldn't that be a separate class? In this case a class LocalMachine with no data_disk_gb.

As long as a class has this variable, we need to check if data_disk_gb is 0 or less.

So, we can create a new class or keep this one but with this check. Even if it does not make sense to a local machine asking for a disk with 0 or negative GB should give an exception.

Reply:
Yes, I think we will need to refactor these classes -- at the moment they are quite messy...
It's called "BaseMachineGroup" which would imply is almost an Abstract class without implementations, but it's full of logic. It's docstring says class  """Base class to manage Google Cloud resources."""  and then takes an argument called "provider" which can be something different than GCP... It's full of if statements regarding the provider, which potentially could be dealt with using inheritance, like Paulo suggested, or leaving some logic to the backend...

@rodrigotuna : you could probably be unblocked by just passing any dummy value for this variable for now --- but I will open an issue for us to refactor these classes.